### DISCUSSION TITLE
Logql query indentation

### DISCUSSION THREAD

**Discussion by rrva** [grafana/grafana#45353] [Dashboards, Panels & Visualizations]
**Is your feature request related to a problem? Please describe.**

I'm always frustrated when I write long logql queries and loose track of parentheses or syntax

**Describe the solution you'd like**

An auto-indent button which indents a query into multiple lines for easier readability

**Describe alternatives you've considered**

Indenting manually

**Additional context**

<img width="1195" alt="Screenshot 2022-02-09 at 11 54 15" src="https://user-images.githubusercontent.com/887132/153184084-d2806e81-52bf-4185-9e32-5249752bdd0a.png">

vs

<img width="1330" alt="Screenshot 2022-02-09 at 11 33 16" src="https://user-images.githubusercontent.com/887132/153182670-6c6f5c99-3597-4fa7-8275-6a7d0caf0e56.png">

@dannykopping also wants this really badly.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
